File: Domotización - Interfaces/Domotización - Interfaces/VentanaAutomatica.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domotización___Interfaces
{
    internal class VentanaAutomatica : Ventana
    {
        public VentanaAutomatica(int numVentana) : base(numVentana)
        {
        }

        public override void bloquear()
        {
            Bloqueada = true;
            Console.WriteLine($"Ventana número {NumVentana} bloqueada automaticamente.");
        }

        public override void desbloquear()
        {
            Bloqueada = false;
            Console.WriteLine($"Ventana número {NumVentana} desbloqueada automaticamente.");
        }
    }
}

File: DomotizacionInterfaces/src/main/java/org/example/VentanaAutomatica.java
package org.example;

public class VentanaAutomatica  extends Ventana{

    protected VentanaAutomatica(int numPuerta) {
        super(numPuerta);
    }

    public void bloquear()
    {
        setBloqueada(true);
        System.out.println("Ventana número " + getNumVentana() + " bloqueada automáticamente");
    }

    public void desbloquear()
    {
        setBloqueada(false);
        System.out.println("Ventana número " + getNumVentana() + " desbloqueada automáticamente.");
    }

    public void Timeout()
    {
        if (!isBloqueada())
        {
            desbloquear();
        }
        else { bloquear(); }
    }

}
